### DISCUSSION TITLE
Can I use code-server to work on a flutter project?

### DISCUSSION THREAD

**Discussion by sinojelly** [coder/code-server#2569] [Q&A]
I tried to install flutter  and dart plugins, but without success.
Any help would be appreciated!

**Comment by code-asher** [comment]
Do you have the flutter sdk and dart installed? The extensions will
require those to be installed on the machine.

**Reply by sinojelly** [reply]
@code-asher @sr229  Thank you all!

I tried to install dart sdk,:

Enviroment:
Android Pad
VHEditor (based on termux and code-server)

Following these steps to install dart sdk:
1. git clone flutter project to local directory.
2. run flutter,   (it says /usr/bin/env does not exists, I changed it to /data/data/vn.vhn.vsc/files/usr/bin/env, then it can run.)
3. it downloads dart-sdk automatically.( but it is x86-64 version, so I remove it, download arm64 manually, unzip it)
4. then i run ./dart in the terminal. It says No such file or directory.

I think maybe some so files dart depends on can not be found.

Is there any solution to this problem? Thanks!

![Screenshot_20210116_094125_vn vhn vsc](https://user-images.githubusercontent.com/520711/104794190-fa62c580-57e0-11eb-93ce-82c15e4b12f6.jpg)

**Reply by code-asher** [reply]
Maybe `/lib/ld-linux-aarch64.so.1` doesn't exist? I'm not sure. What are you running this on?

**Reply by code-asher** [reply]
Oh wait you said Android. I think the binary needs to be built specifically for Android to work. Have you tried installing it through Termux? That might work.

**Reply by sr229** [reply]
> Oh wait you said Android. I think the binary needs to be built specifically for Android to work. Have you tried installing it through Termux? That might work.

the said app seems to be based on Termux, as Termux libraries do not map out to `/lib/ld-linux` (they are built using the NDK), they'll need a proper chroot for this.

**Reply by code-asher** [reply]
It doesn't look like flutter is available but it does have dart:

```
pkg install dart
```

That said, I haven't used it so who knows if it works.

**Reply by sr229** [reply]
> It doesn't look like flutter is available but it does have dart:
> 
> ```
> pkg install dart
> ```
> 
> That said, I haven't used it so who knows if it works.

you have to install the Flutter SDK by its own, Dart itself is just the runtime.

**Reply by code-asher** [reply]
Yup.

**Comment by sr229** [comment]
Hey there, the Flutter plugins should be available on the marketplace. VSCode supports this, but make sure you have the SDK installed firsthand.

**Comment by bilogic** [comment]
@sinojelly did you get anywhere with this?
